### PR TITLE
Add documentation for Grafana monitoring for ChatQnA microservices

### PR DESCRIPTION
**Pull request by ctao456**
Initial writeup of grafana monitoring, subject to changes over time.

### CODE REVIEW

File: examples/ChatQnA/deploy/monitoring.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+
+# Monitoring ChatQnA deployment
+
+Now that you have deployed the ChatQnA example, let's talk about monitoring the performance of the microservices in the ChatQnA pipeline.
+
+Monitoring the performance of microservices is crucial for ensuring the smooth operation of the generative AI systems. By monitoring metrics such as latency and throughput, you can identify bottlenecks, detect anomalies, and optimize the performance of individual microservices. This allows us to proactively address any issues and ensure that the ChatQnA pipeline is running efficiently.
+
+This document will help you understand how to monitor in real time the latency, throughput, and other metrics of different microservices. You will use **Prometheus** and **Grafana**, both open-source toolkits, to collect metrics and visualize them in a dashboard.
+
+### Set up the Prometheus server
```
Comment (dbkinder) [inline examples/ChatQnA/deploy/monitoring.md:10]: You shouldn't skip from the H1 (title) to an H3 heading.  This should be an H2 heading (with two leading ##).

File: examples/ChatQnA/deploy/monitoring.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+Prometheus is a tool used for recording real-time metrics and is specifically designed for monitoring microservices and alerting based on their metrics.
+
+The `/metrics` endpoint on the port running each microservice exposes the metrics in the Prometheus format. The Prometheus server scrapes these metrics and stores them in its time series database. For example, metrics for the Text Generation Interface (TGI) service are available at:
+```bash
+http://${host_ip}:9009/metrics 
+```
+
+To set up the Prometheus server, follow these steps:
+
+#### 1. Download Prometheus:
```
Comment (dbkinder) [inline examples/ChatQnA/deploy/monitoring.md:21]: ... and all these numbered titles should be one level up (### not ####).  Also all these headings shouldn't end with `:`

File: examples/ChatQnA/deploy/monitoring.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+>Note: Before starting Prometheus, ensure that no other processes are running on the designated port (default is 9090). Otherwise, Prometheus will not be able to scrape the metrics.
+
+On the Prometheus UI, you can see the status of the targets and the metrics that are being scraped. You can search for a metrics variable by typing it in the search bar.
+
+The TGI metrics can be accessed at:
+```bash
+http://${host_ip}:9009/metrics 
+```
+
+### Set up the Grafana dashboard
```
Comment (dbkinder) [inline examples/ChatQnA/deploy/monitoring.md:82]: This should be back to an H2 (##)

File: examples/ChatQnA/deploy/monitoring.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+Grafana is a tool used for visualizing metrics and creating dashboards. It can be used to create custom dashboards that display the metrics collected by Prometheus.
+
+To set up the Grafana dashboard, follow these steps:
+#### 1. Download Grafana:
```
Comment (dbkinder) [inline examples/ChatQnA/deploy/monitoring.md:86]: And all these should be H3 (###)

**Review by dbkinder — CHANGES_REQUESTED** [review]
All this content is a duplication of what's already in examples/ChatQnA/ChatQnA_Guide.rst.  Why are we creating a new doc in this PR with the same content?

**Comment by mkbhanda** [comment]
Oh no, sad about duplicate effort.
Should we put graphana Prometheus etc in a central place and refer to it via link everywhere
________________________________

**Comment by ctao456** [comment]
> All this content is a duplication of what's already in examples/ChatQnA/ChatQnA_Guide.rst. Why are we creating a new doc in this PR with the same content?

Hi @dbkinder. I checked history of examples/ChatQnA/ChatQnA_Guide.rst, and found this [commit](https://github.com/opea-project/docs/commit/44ab3aa3f7d988bdec7bb14f3c3781da374404d2) by @tomlenth and @mkbhanda that already included this content. Indeed, this PR is probably not needed.

**Comment by ctao456** [comment]
> Oh no, sad about duplicate effort. Should we put graphana Prometheus etc in a central place and refer to it via link everywhere

@mkbhanda There was not too much duplicated effort. I drafted the writeup of this monitoring section, and it's already pushed to examples/ChatQnA/ChatQnA_Guide.rst. Submitting this PR didn't take me a long time.

**Comment by dbkinder** [comment]
Should we close this PR since the content is already in examples/ChatQnA/ChatQnA_Guide.rst ?

**Comment by ctao456** [comment]
> Should we close this PR since the content is already in examples/ChatQnA/ChatQnA_Guide.rst ?

Yes, David. Let's close this PR. Thank you for the comments on headings.

**Pull request closed without merging**